Peer review

oahray opened the pull request:
Please leave feedback.

oahray committed:
Add template pages

oahray committed:
Add folder for lib files

oahray committed:
Modify message board

oahray committed:
Modify message board

oahray committed:
Modified groups tab of userboard

oahray committed:
Update README.md

oahray committed:
Update README.md

oahray committed:
UI modifications based on BFA feedback

oahray committed:
Merge pull request #1 from oahray/issues

UI modifications based on BFA feedback

oahray committed:
Set up models and migration files

oahray committed:
Transpile server es6 code into build folder

oahray committed:
Add POST /api/group folder

oahray committed:
Set up sequelize class and instance methods

oahray committed:
Add Express sessions and DELETE /api/user/logout route

oahray committed:
Make some users route private

oahray committed:
Fix issues with POST /api/group route

oahray committed:
Add middleware to check if current user belongs to specified group

oahray committed:
Add username character validation before DB query

oahray committed:
Modify package.json to run es6"

oahray committed:
Set server to run with babel-node

oahray committed:
Set server to run with babel-node

oahray committed:
Modify package.json"

oahray committed:
1.0.1

oahray committed:
Set heroku to work with babel-node

oahray committed:
Fix bug with creating messages

oahray committed:
Set up app to use a different heroku postgres"

oahray committed:
Refactor code to es6 standard

oahray committed:
Add error handler for isTaken middleware

oahray committed:
Modify package.json with postinstall script"

oahray committed:
Modify package.json with postinstall script"

oahray committed:
Modify package.json with postinstall script"

oahray committed:
Add sequelize-cli as devDependency to fix postinstall issues"

oahray committed:
Add new npm script"

oahray committed:
Add sequelize-cli as dependency to fix heroku db:migrate issues"

oahray committed:
Re-add config.json"

oahray committed:
Change message priority type from ENUM to STRING

oahray committed:
Add travis.yml

oahray committed:
Fix bug in models index.js file causing app to crash

oahray committed:
Merge pull request #2 from oahray/api

Api

oahray committed:
Modify readMe

oahray committed:
Merge pull request #3 from oahray/readme

Modify readMe

oahray committed:
Add info route for API, Modify frontend

oahray committed:
Merge pull request #4 from oahray/api-fix

Add info route for API, Modify frontend

oahray committed:
Add tests for signin and signup

oahray committed:
Merge pull request #5 from oahray/app-tests

Add tests for signin and signup

oahray committed:
Fix config.json for production

oahray committed:
Add more tests

oahray committed:
Merge pull request #6 from oahray/more-tests

Add more tests

oahray committed:
Specify stable for nodejs version

oahray committed:
Merge pull request #7 from oahray/travis

Specify stable for nodejs version

oahray committed:
Specify new branch for travis

oahray committed:
Specify remove sequelize db migrate script causing Travis to fail

oahray committed:
Remove before_script commands causing Travis fail

oahray committed:
Specify remove sequelize db migrate script causing Travis to fail

oahray committed:
Specify remove sequelize db migrate script causing Travis to fail

oahray committed:
Specify remove sequelize db migrate script causing Travis to fail

oahray committed:
Add tests for models

oahray committed:
Modify travis.yml

oahray committed:
Modify travis.yml before_script

oahray committed:
Modify travis.yml

oahray committed:
Modify test before hooks

oahray committed:
Modify test before hooks

oahray committed:
Modify tests for travisci

oahray committed:
Add env variable to travis.yml

oahray committed:
Modify tests with before hooks

oahray committed:
Merge pull request #8 from oahray/travis

Travis

oahray committed:
Update README.md

oahray committed:
Add routes to rename group, change password and change email

oahray committed:
Add new routes to modify group properties and leave group

oahray committed:
Merge pull request #9 from oahray/more-api-routes

More api routes

oahray committed:
Add signup and signin components

oahray committed:
Add components and fix major bugs

oahray committed:
Fix SideBar bugs

oahray committed:
Fix component not rerendering when url changes, UI fix

oahray committed:
Fix Create Group bug

oahray committed:
Modify user search and add user

oahray committed:
Fix pagination in search window

oahray committed:
Modify webpack production config

oahray committed:
Modify heroku-postbuild script

oahray committed:
Modify dependencies list in package.json to fix postbuild error

oahray committed:
Add missing babel plugins to package.json

oahray committed:
Modify package.json

oahray committed:
Modify package.json

oahray committed:
Modify package.json

oahray committed:
Fix Redux-Thunk error

oahray committed:
Fix Cors error

oahray committed:
Modify dependencies list after heroku crash

oahray committed:
Modify base url

oahray committed:
Modify material icons link in html

oahray committed:
Fix viewport

oahray committed:
Modify migrations for usergroups

oahray committed:
Modify migrations

oahray committed:
Modify api for readBy feature

oahray committed:
fix message input bug

oahray committed:
fix message input bug

oahray committed:
fix message input bug

oahray committed:
fix message input bug

oahray committed:
fix message input bug

oahray committed:
Merge pull request #10 from oahray/materialize

Materialize

oahray committed:
Fixed group component display

oahray committed:
Merge pull request #11 from oahray/ui-fix

Fixed group component display

oahray committed:
Add real-time update of group users with socket

oahray committed:
Merge pull request #12 from oahray/ft-include-socketIo-151004344

Add real-time update of group users with socket

oahray committed:
Fix bug that caused errors on production, added tests

oahray committed:
fix styling issues

oahray committed:
Merge pull request #13 from oahray/bg-fix-production-bug-151084530

Bg fix production bug 151084530

johnkingzy commented on client/src/containers/Group.jsx:
```diff
@@ -0,0 +1,262 @@
+  }
+
+  scrollToBottom() {
+    // this.messageList.scrollTop = this.messageList.scrollHeight;
```
All lines within method has benn commented out, can you take this off

veryobinna commented on client/src/actions/auth/requestReset.js:
```diff
@@ -0,0 +1,108 @@
+  };
+};
+
+export const requestReset = (email) => {
```
You need to add JSDocs to all functions.

veryobinna commented on client/src/actions/auth/signinAction.js:
```diff
@@ -0,0 +1,52 @@
+};
+
+export const signinUser = (username, password) => {
+  return (dispatch) => {
```
You're passing dispatch to the anonymous function, you seem not to make use of the argument.

veryobinna commented on client/src/actions/auth/signupAction.js:
```diff
@@ -0,0 +1,54 @@
+    })
+    .catch((err) => {
+      if (err.response) {
+        console.log(err.response.data);
```
No console logs

veryobinna commented on client/src/actions/group/getGroupsList.js:
```diff
@@ -0,0 +1,51 @@
+
+export const getGroupList = (token) => {
+  const FETCH_URL = `${BASE_URL}/user/me/groups`;
+  return (dispatch, getState) => {
```
are the arguments necessary?

veryobinna commented on client/src/actions/group/searchUsers.js:
```diff
@@ -0,0 +1,129 @@
+};
+
+
+// ADD USER FROM SEARCH RESUKTS
```
You can do better docs with JSDocs

veryobinna commented on client/src/actions/group/searchUsers.js:
```diff
@@ -0,0 +1,129 @@
+  };
+};
+
+// export const resetAddUserStatus = () => {
```
Remove unnecessary code

veryobinna commented on client/src/containers/App.jsx:
```diff
@@ -0,0 +1,87 @@
+  }
+
+  componentWillMount() {
+    if (window.localStorage && window.localStorage.getItem('x-auth')) {
```
window.localstorage.getItem() is fine. You can remove the window.localstorage.

veryobinna commented on client/src/containers/Group.jsx:
```diff
@@ -0,0 +1,262 @@
+
+  componentWillReceiveProps(newProps) {
+    if (this.props.selectedGroup && Number(newProps.match.params.groupid) !== this.props.selectedGroup.id) {
+      console.log(typeof newProps.match.params.groupid);
```
no console logs

veryobinna commented on client/src/containers/Group.jsx:
```diff
@@ -0,0 +1,262 @@
+    );
+
+    const messagesTab = (
+      this.state.messageOpen ? showOpenMessage : messages
```
don't set state directly, use the setState method

veryobinna commented on client/src/containers/UserHome.jsx:
```diff
@@ -0,0 +1,80 @@
+  }
+}
+
+function mapDispatchToProps(dispatch) {
```
all functions should be in es6

veryobinna commented on client/src/reducers/createdGroup.js:
```diff
@@ -0,0 +1,12 @@
+
+export default (state = null, action) => {
+  switch (action.type) {
+    case CREATE_GROUP_SUCCESS:
```
Don't you think state will be mutated here? Make use of es6 spread operator to add new state.

veryobinna replied:
Do the  same for all reducers

veryobinna commented on coverage/bc-24-postit/server/models/group.js.html:
```diff
@@ -0,0 +1,164 @@
+<!doctype html>
```
Why do you have HTML on the backend path?

johnkingzy commented on server/routes/index.js:
```diff
@@ -0,0 +1,93 @@
+  app.get('/api/group/:groupid/users', authenticate,
+  isGroupMember, groupsController.getGroupUsers);
+
+  app.post('/api/group/:groupid/message', authenticate,
```
I see you are currently using express v4.x which provides you with a router instance to chain endpoint of different methods. Use this as a guide 

https://www.codementor.io/johnkingzy/api-routing-using-expressjs-router-4-0-9yohet6uu

johnkingzy commented on server/middleware/isTaken.js:
```diff
@@ -0,0 +1,30 @@
+      });
+    }
+    User.findOne({ where: { email } }).then((userAgain) => {
+      if (userAgain) {
```
What do you mean by user again?

johnkingzy commented on server/middleware/isGroupMember.js:
```diff
@@ -0,0 +1,34 @@
+
+export default (req, res, next) => {
+  const groupId = req.params.groupid;
+  if (!Number(groupId)) {
```
Instead of using the Javascript Number Object, try `isNaN()`

johnkingzy commented on server/controllers/users.js:
```diff
@@ -0,0 +1,296 @@
+// Function to signup new users
+export const signup = (req, res) => {
+  const username = req.body.username.trim().toLowerCase();
+  const email = req.body.email.trim().toLowerCase();
```
What if I pass in an invalid email address for example `emailaddress`? I can't see where you covered that scenario.

johnkingzy commented on server/controllers/users.js:
```diff
@@ -0,0 +1,296 @@
+  const username = body.username.trim().toLowerCase();
+  if (!body.password) {
+    return res.status(400).json({
+      error: 'Password must not be empty'
```
I will suggest you try as much as possible to be consistent with your design pattern, I can see you are using a middleware to validate the username, but you seem to be validating the password inside controller.

johnkingzy commented on server/controllers/users.js:
```diff
@@ -0,0 +1,296 @@
+import * as _ from 'lodash';
+import randomstring from 'randomstring';
+import { User, Message } from '../models';
```
Leave a blank line between relative imports and absolute imports

johnkingzy commented on server/app.js:
```diff
@@ -0,0 +1,75 @@
+const compiler = webpack(webpackConfig);
+
+const env = process.env.NODE_ENV || 'development';
+console.log('>>>>>> ENV: ', env);
```
I feel you might want to take this out, `console.log`

johnkingzy commented on server/app.js:
```diff
@@ -0,0 +1,75 @@
+dotenv.config();
+
+if (env === 'development') {
+  dotenv.config();
```
i don't think you need to repeat `dotenv.config()` twice.

johnkingzy commented on server/app.js:
```diff
@@ -0,0 +1,75 @@
+app.use('/', express.static(publicPath));
+
+// Require our routes
+require('./routes')(app);
```
I will suggest you are consistent your with coding style, I can see you are using es6 module system which is `import/export` but right here you happen to be using the commonjs module system which is `require` here.

johnkingzy commented on server/config/config.js:
```diff
@@ -0,0 +1,27 @@
+  },
+  production: {
+    username: 'tpgpwvuylmgwfq',
+    password: '721df25847ce113ee7306f1c673d092b788b53ea48ad8b75f797c25695f912b8',
```
your private info is visible, try securing this as well.

johnkingzy commented on client/src/reducers/setUserReducer.js:
```diff
@@ -0,0 +1,20 @@
+  SIGNIN_SUCCESS, LOGOUT_USER, RESET_PASSWORD_SUCCESS } from '../actions';
+
+export default function (state = null, action) {
+  console.log('>>>>>>>>>>>>>> User Reducer Triggered! by:', action);
```
I think you should take this off

johnkingzy commented on client/src/containers/UserHome.jsx:
```diff
@@ -0,0 +1,80 @@
+  }
+}
+
+function mapStateToProps(state) {
```
Missing JSDoc for function

oahray committed:
Refactor signin and signup action creators

oahray committed:
Set .travis.yml branches

oahray committed:
Add sequelize-cli to package.json to fix travis failure

oahray committed:
Add custom migrate script .travis.yml

oahray committed:
Add nyc to package dev dependencies

oahray committed:
Remove quotes from .travis.yml

oahray committed:
Set travis.yml env as global

oahray committed:
fix test issues

oahray committed:
fix test failing on travis

oahray committed:
chore(feedback):
- Remove console.log
- Modify UI
- Add more tests
- Add JSdocs

oahray committed:
Fix test failing on Travis

oahray committed:
Fix test failing on Travis

oahray committed:
Merge pull request #15 from oahray/ch-implement-feedback-151163001

Ch implement feedback 151163001

oahray committed:
Fix coveralls.yml

oahray committed:
Add blank line to coverall.yml

oahray committed:
Merge pull request #16 from oahray/ch-implement-feedback-151163001

Fix coveralls.yml

oahray committed:
Fix documentation link in README

oahray committed:
Merge pull request #17 from oahray/ch-implement-feedback-151163001

Fix documentation link in README

oahray committed:
ch(feedback): implement feedback
- fix full page reload on delete group
-fix spinner load on new message
- Fix styling

oahray committed:
Merge pull request #18 from oahray/ch-implement-feedback-151163001

ch(feedback): implement feedback

oahray committed:
chore(tests): write tests for client side
[finishes #153260501]

oahray committed:
ch(feedback): implement feedback
- Fix issues with material modal initialization
- Fix error sending mail on production

oahray committed:
log nodemailer error on fail

oahray committed:
bg(email notification): fix email not sending on hosted app

oahray committed:
bg(email notificataion): permit non-secure apps on gmail

oahray committed:
ch(readme): modify readme
- Add required readme points
- Add licence
- Add .sample.env

oahray committed:
Fix missing codeclimate badge

oahray committed:
configure codeclimate.yml

oahray committed:
Merge pull request #19 from oahray/implement-feedback

Implement feedback

oahray committed:
Add client folder to codeclimate

oahray committed:
Merge pull request #20 from oahray/implement-feedback

Add client folder to codeclimate

oahray committed:
ch(feedback): Add tests

oahray committed:
ch(feedback): Fix isGroupMember test failing on travis

oahray committed:
modify middleware.test.js to debug fail on travis

oahray committed:
bg(feedback): tests fix

oahray committed:
Merge pull request #21 from oahray/implement-feedback

Implement feedback

oahray committed:
ch(feedback): pipe travis test results to coveralls

oahray committed:
Merge pull request #22 from oahray/implement-feedback

ch(feedback): pipe travis test results to coveralls

oahray committed:
ch(feedback): Add js docs

oahray committed:
Chore(maintain): Refactor code and fix issues

oahray committed:
Merge pull request #23 from oahray/implement-feedback

Implement feedback

oahray committed:
BugFix(task-runner) modify webpack production config
- add 'define plugin' properties for cloudinary parameters

oahray committed:
fix styling for edit profile page

oahray committed:
Chore(maintain): Optimise for mobile
- fix modal min height
- make profile images responsive

oahray committed:
Merge pull request #24 from oahray/implement-feedback

Implement feedback

oahray committed:
Revert "Implement feedback"

oahray committed:
Merge pull request #26 from oahray/revert-24-implement-feedback

Revert "Implement feedback"

oahray committed:
Revert "Implement feedback"

oahray committed:
Merge pull request #28 from oahray/revert-23-implement-feedback

Revert "Implement feedback"

oahray committed:
Revert "Revert "Implement feedback""

oahray committed:
Merge pull request #29 from oahray/revert-28-revert-23-implement-feedback

Revert "Revert "Implement feedback""

oahray committed:
Revert "Revert "Implement feedback""

oahray committed:
Merge pull request #30 from oahray/revert-26-revert-24-implement-feedback

Revert "Revert "Implement feedback""

oahray committed:
fix merge conflict

oahray committed:
Merge pull request #25 from oahray/chore/153260501/client-side-tests

Tests to be completed shortly

oahray committed:
chore(tests): Add tests for more components

oahray committed:
Chore(refactor): refactor code and modify readme

oahray committed:
Chore(refactor): rename www file in /server/bin folder

oahray committed:
chore(tests): Finish client side tests
- finish tests for containers

oahray committed:
Merge pull request #31 from oahray/chore/153260501/client-side-tests

#153260501 Add tests for more components

oahray committed:
chore(tests): Write end-to-end tests

oahray committed:
chore(tests): end-to-end tests

oahray committed:
chore(refactor): refactor code to adhere with requirements

oahray committed:
Merge pull request #32 from oahray/chore/153566891/e2e-tests

#153566891 e2e tests

oahray committed:
chore(feedback): implement feedback
- Add JSDoc comments to remaining functions
- Fix issue with long emails displaced
- Fix issues with images overflowing container
- Refactor tests
- Modify README to include more sections and add context
- Modify API Docs
- Remove obsolete npm scripts
- Fix issue with 401 error in console when not logged in
- Discontinue use of 'data' as parameter name
[finishes #153673866]

oahray committed:
Merge pull request #33 from oahray/chore/153673866/implement-feedback

chore(feedback): implement feedback

oahray committed:
bugfix(UI): Style welcome text in landing page

oahray committed:
Merge pull request #34 from oahray/bugfix/153673866/ui-fixes

#153673866 Style welcome text in landing page
